packages/jest-matchers/src/index.js

```diff
@@ -38,32 +38,32 @@ const {
 const GLOBAL_STATE = Symbol.for('$$jest-matchers-object');
 
 class JestAssertionError extends Error {
-  matcherResult: any
+  matcherResult: any;
 }
 
 if (!global[GLOBAL_STATE]) {
-  Object.defineProperty(
-    global,
-    GLOBAL_STATE,
-    {value: {
+  Object.defineProperty(global, GLOBAL_STATE, {
+    value: {
       matchers: Object.create(null),
       state: {
         assertionsExpected: null,
         assertionsMade: 0,
         suppressedErrors: [],
       },
-    }},
-  );
+    },
+  });
 }
 
 const expect: Expect = (actual: any): ExpectationObject => {
   const allMatchers = global[GLOBAL_STATE].matchers;
   const expectation = {not: {}};
   Object.keys(allMatchers).forEach(name => {
-    expectation[name] =
-      makeThrowingMatcher(allMatchers[name], false, actual);
-    expectation.not[name] =
-      makeThrowingMatcher(allMatchers[name], true, actual);
+    expectation[name] = makeThrowingMatcher(allMatchers[name], false, actual);
+    expectation.not[name] = makeThrowingMatcher(
+      allMatchers[name],
+      true,
+      actual,
+    );
```

Comment:
We are trying to avoid at all cost to do those kind of conditionals inside of prettier. The issue is that the runtime is quadratic for them because you have to do them again recursively.

The way prettier algorithm is structured is that there are groups and they can either fit in one line or break. If they break, we can add indent and insert characters, but that's it.

We could implement the behavior you mention but I feel like it is more consistent to always use a single way to break rather than multiple and therefore not worth the extra complexity.